include/devicetree/pwms.h

```diff
@@ -364,6 +437,41 @@ extern "C" {
  */
 #define DT_INST_PWMS_LABEL(inst) DT_INST_PWMS_LABEL_BY_IDX(inst, 0)
 
+/**
+ * @brief Get a pointer to the pwm-controller device object
+ *        from a DT_DRV_COMPAT instance's pwms by index
+ *
+ * @param inst DT_DRV_COMPAT instance number
+ * @param idx logical index into pwms property
+ * @return The node identifier to the pwm-controller of the node referenced
+ *         at index "idx"
+ * @see DT_PWMS_CTLR_BY_IDX()
+ */
+#define DT_INST_PWMS_CTLR_BY_IDX(inst, idx) \
+	DT_PWMS_CTLR_BY_IDX(DT_DRV_INST(inst), idx)
+
+/**
+ * @brief Get a pointer to the pwm-controller device object
+ *        property by name
+ * @param inst DT_DRV_COMPAT instance number
+ * @param name lowercase-and-underscores name of a pwms element
+ *             as defined by the node's pwm-names property
+ * @return The node identifier of the pwm-controller referenced
+ *         at the named element
+ * @see DT_PWMS_CTLR_BY_NAME()
+ */
+#define DT_INST_PWMS_CTLR_BY_NAME(inst, name) \
+	DT_PWMS_CTLR_BY_NAME(DT_DRV_INST(inst), name)
+
+/**
+ * @brief Equivalent to DT_INST_PWMS_CTLR_BY_IDX(inst, 0)
+ * @param inst DT_DRV_COMPAT instance number
+ * @return The node identifier of the pwm-controller device object
+ *         referenced at index 0
+ * @see DT_PWMS_CTLR_BY_IDX()
+ */
+#define DT_INST_PWMS_CTLR(inst) DT_INST_PWMS_LABEL_BY_IDX(inst, 0)
```

Comment:
```suggestion
#define DT_INST_PWMS_CTLR(inst) DT_INST_PWMS_CTLR_BY_IDX(inst, 0)
```

This one looks like it doesn't work.